src/hermes/commands/deposit/base.py

```diff
@@ -21,8 +19,7 @@ class BaseDepositPlugin(HermesPlugin):
     TODO: describe workflow... needs refactoring to be less stateful!
     """
 
-    def __init__(self, click_ctx: click.Context, ctx: CodeMetaContext) -> None:
-        self.click_ctx = click_ctx
+    def __init__(self, ctx):
         self.ctx = ctx
```

Comment:
**thought:** We might decide to keep the ``HermesContext`` and derived mischief for now. I see the need to refactor the "deposit" (base) plugin which is heavily state dependent. However, we are doing data processing, hence I think it's a good idea to reflect that in the design...

However, to go forward I think it's easiest to keep the ctx stuff for now, design the plugin interface in a way that only the "Command" does interact with it and have a better hinge point for later refactoring.

To sum it up: For now, do the refactoring using context, design new plugins that the data is visible, later get rid of contexts.
Disclaimer: The purpose of this note is to derive an issue out of it ;)